ExecuteWithConnectionAsync追加 close #13

huurou opened the pull request:
Testも追加したが現状使用例を示すためのもの
#16でGetCommandの引数からconnectionを削除しましたが、これは間違いだったので修正しました。

huurou committed:
ExecuteWithConnectionAsync追加 close #13

- Testも追加したが現状使用例を示すためのもの
- #16でGetCommandの引数からconnectionを削除したのを修正

ronron1114 commented on LibDbHelper.Test/DbHelperTest.cs:
```diff
@@ -245,6 +245,33 @@ public async Task ExecuteAsync()
+
+            // Assert
+        }
+
```
コミットかロールバックしたらコネクションってとじなくても大丈夫なんだっけ

huurou replied:
コネクションは閉じるかusingを使用すべきですね…
usingを使用するよう修正します。

huurou replied:
修正したので解決済とします。

ronron1114 commented on LibDbHelper/DbHelper.cs:
```diff
@@ -20,9 +20,9 @@ protected DbHelper(string connectionString)
         }
 
-        protected abstract DbConnection GetConnection();
+        public abstract DbConnection GetConnection();
```
外でOpenできるようになっちゃったけどしょうがないですか；；

huurou replied:
しょうがないですね。あくまでスコープが最低限になるようにprotectedにしていただけなので、コネクションを引数に取れるようにするならpublicにする必要があります。コネクションを引数に取らないようにする等の方法を考える必要があります。

ronron1114 replied:
わかりました！

ronron1114 commented on LibDbHelper/DbHelper.cs:
```diff
@@ -20,9 +20,9 @@ protected DbHelper(string connectionString)
 
-        protected abstract DbCommand GetCommand(string sql);
+        protected abstract DbCommand GetCommand(string sql, DbConnection connection);
 
```
個人的にconnection → sqlのほうが好みですけどいかがですか

huurou replied:
いいですね。そうしましょう。

huurou replied:
OracleCommandが`cmdText, conn`という順番だったのでそれに合わせていたんでした。
```c#
public OracleCommand(string cmdText, OracleConnection conn)
```
どうしますか。

ronron1114 replied:
NpgsqlCommandもその順番ですね。そういうもんなんかって感じなので変更なしでお願いします！

huurou committed:
ExecuteWithConnectionAsync修正

- メソッド内でOpenしていた部分を削除 複数回呼び出されるとすでにOpen済のコネクションを再度開くことになってしまう。使用する際はOpen済のコネクションを渡してもらう
- テスト内でコネクションインスタンス作成にusingステートメントを追記
